### PR TITLE
Implement "Sentinels list automatic refresh"

### PR DESCRIPTION
**Pull request by dbackeus**
As specified in the Sentinel client spec (https://redis.io/docs/reference/sentinel-clients):

```markdown
Optionally once a successful reply to get-master-addr-by-name is received, a client may update 
its internal list of Sentinel nodes following this procedure:

- Obtain a list of other Sentinels for this master using the command SENTINEL sentinels <master-name>.
- Add every ip:port pair not already existing in our list at the end of the list.
```

Unfortunately I'm having some issues running the test suite via `rake` ~~so not sure how to proceed with testing without getting CI going~~. The first issue I'm getting is `Waiting for redis (port 16380)... timeout.` at the start of the run which breaks pretty much everything. If I manually run `redis-server --port 16380` most of the tests pass but I'm still left with 26 errors. On the plus side my changes do not appear to add any additional errors 😅 

I'd love some feedback on this. Are you interested in accepting this feature? What are you expecting with regards to tests? Do you feel like the feature should be optional / configurable, if so: opt-in or opt-out? (personal take: this should be a good idea for most cases, if optionality is desired I would suggest allowing explicit opt-out)

### CODE REVIEW

**Comment by dbackeus** [comment]
Thanks for enabling CI. Will take a lot at fixing the failing tests tomorrow.

**Comment by byroot** [comment]
> The first issue I'm getting is Waiting for redis (port 16380)... timeout. at the start of the run which breaks pretty much everything.

Hum, you should look in `log/redis.log`, there might the be reason why it's not responding?

> Are you interested in accepting this feature? 

Yes.

> What are you expecting with regards to tests? 

Just some basic coverage, simulate that refresh scenario. It's been a while since I wrote the sentinel tests, but there's some basic mocking primitives that should allow to simulate this without too much work?

> Do you feel like the feature should be optional / configurable, if so: opt-in or opt-out?

I'll have to read on it a bit, but I don't see a reason not to do it? Except maybe the extra queries on reconnect, but it's probably fine.

**Comment by dbackeus** [comment]
@byroot implemented a test to verify that a successful connection refreshes the list. Marking this as ready for review. Let me know if anything needs changing or feel free to modify the PR yourself.

File: lib/redis_client/sentinel_config.rb [hunk 1]
```diff
@@ -159,5 +165,19 @@ def each_sentinel
 
       raise last_error if last_error
     end
+
+    def refresh_sentinels(sentinel_client)
+      sentinel_response = sentinel_client.call("SENTINEL", "sentinels", @name, &@to_list_of_hash)
+      sentinels = sentinel_response.map do |sentinel|
+        { host: sentinel.fetch("ip"), port: sentinel.fetch("port").to_i }
```
Comment (casperisfine) [inline lib/redis_client/sentinel_config.rb:172]:
```suggestion
        { host: sentinel.fetch("ip"), port: Integer(sentinel.fetch("port")) }
```

For consistency and strictness. Shouldn't do any difference in practice.

**Review by casperisfine — APPROVED** [review]
One very minor nitpick, I'll fix it on master.

**Comment by casperisfine** [comment]
Ref: https://github.com/redis-rb/redis-client/issues/107

So this introduced a bug for sentinels that need a password. I'll try to figure out a fix.

**Pull request merged**